nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -57,7 +57,7 @@ inline BT::NodeStatus GoalUpdater::tick()
 
   getInput("input_goal", goal);
 
-  callback_group_executor_.spin_some();
```

Comment:
This fixes the goal_updater test. The problem is that for some reason with `spin_some()`, even if there is theoretically some work in the pipeline, the callbacks are not called. Interestingly enough, the test also succeeds if we call `spin_some` twice in a row

```
callback_group_executor_.spin_some();
callback_group_executor_.spin_some();
```

I've also check that this fails in other tests like `test_is_battery_charging` but the overall test succeeds coincidentally because FAILURE is expected whether or not the subscription callback is called.

I don't think this is a recent issue, I've seen this in the past. I'd propose in this PR to just fix what is obviously broken and create a separate ticket to check whether or not `spin_some()` triggers the callbacks

Reply:
Worth noting to @clalancette